slides/kube/rollout.md

```diff
@@ -94,6 +94,26 @@ That rollout should be pretty quick. What shows in the web UI?
 
 ---
 
+## Give it some time
+
+- At first, it looks like nothing is happening (the graph remains at the same level)
+
+- According to `kubectl get deploy -w`, the `deployment` was updated really quickly
+
+- But `kubectl get pods -w` tells a different story
+
+- The old `pods` are still here, and they stay in `Terminating` state for a while
+
+- Eventually, they are terminated; and then the graph decreases significantly
+
+- This delay is due to the fact that our worker doesn't handle signals
+
+- Kubernetes sends a "polite" shutdown request to the worker, which ignores it
+
+- Eventually, Kubernetes gets impatient and kills the container
```

Comment:
Do we want to clarify what governs the timing for "eventually"? (I'm guessing a setting in the config.)

Reply:
Graceful shutdown has a 30s timeout by default: https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods

I'll clarify too.

Reply:
I added details about the grace period.